arrayfire_wrapper/_backend.py

```diff
@@ -64,40 +70,85 @@ def _get_backend_path_config() -> _BackendPathConfig:
             ctypes.windll.kernel32.SetErrorMode(0x0001 | 0x0002)  # type: ignore[attr-defined]
 
         if not af_path:
-            af_path = _find_default_path(f"C:/Program Files/ArrayFire/v{ARRAYFIRE_VER_MAJOR}")
+            try:
+                af_path = _find_default_path(f"C:/Program Files/ArrayFire/v{ARRAYFIRE_VER_MAJOR}")
+            except ValueError:
+                af_path = None
+
 
         if cuda_path and (cuda_path / "bin").is_dir() and (cuda_path / "nvvm/bin").is_dir():
             cuda_found = True
 
-        return _BackendPathConfig("", ".dll", af_path, cuda_found)
+        return _BackendPathConfig("", ".dll", af_path, af_is_user_path, cuda_found)
 
     if platform_name == _SupportedPlatforms.darwin.value:
         default_cuda_path = Path("/usr/local/cuda/")
 
         if not af_path:
             af_path = _find_default_path("/opt/arrayfire", "/usr/local")
+            try:
+                af_path = _find_default_path(f"C:/Program Files/ArrayFire/v{ARRAYFIRE_VER_MAJOR}",
+                                              "C:/Program Files (x86)/ArrayFire/v{ARRAYFIRE_VER_MAJOR}")
+            except ValueError:
+                af_path = None
 
         if not (cuda_path and default_cuda_path.exists()):
             cuda_found = (default_cuda_path / "lib").is_dir() and (default_cuda_path / "/nvvm/lib").is_dir()
 
-        return _BackendPathConfig("lib", f".{ARRAYFIRE_VER_MAJOR}.dylib", af_path, cuda_found)
+        return _BackendPathConfig("lib", f".{ARRAYFIRE_VER_MAJOR}.dylib", af_path, af_is_user_path, cuda_found)
 
     if platform_name == _SupportedPlatforms.linux.value:
         default_cuda_path = Path("/usr/local/cuda/")
 
         if not af_path:
-            af_path = _find_default_path(f"/opt/arrayfire-{ARRAYFIRE_VER_MAJOR}", "/opt/arrayfire/", "/usr/local/")
+            try:
+                af_path = _find_default_path(f"/opt/arrayfire-{ARRAYFIRE_VER_MAJOR}", "/opt/arrayfire/", "/usr/local/")
+            except ValueError:
+                af_path = None
 
         if not (cuda_path and default_cuda_path.exists()):
             if "64" in platform.architecture()[0]:  # Check either is 64 bit arch is selected
                 cuda_found = (default_cuda_path / "lib64").is_dir() and (default_cuda_path / "nvvm/lib64").is_dir()
             else:
                 cuda_found = (default_cuda_path / "lib").is_dir() and (default_cuda_path / "nvvm/lib").is_dir()
 
-        return _BackendPathConfig("lib", f".so.{ARRAYFIRE_VER_MAJOR}", af_path, cuda_found)
+        return _BackendPathConfig("lib", f".so.{ARRAYFIRE_VER_MAJOR}", af_path, af_is_user_path, cuda_found)
 
     raise OSError(f"{platform_name} is not supported.")
 
+# finds paths to locally packaged arrayfire libraries if they exist in site
+def _find_site_local_path() -> Path:
+    local_paths = ["."]
```

Comment:
You would better make a Path type object and then append objects when you work with this variable, otherwise, this row is completely pointless https://github.com/roaffix/arrayfire-python-wrapper/pull/12/files#diff-9ec796ab58c0b73aa387c4753fc01e558c6a7849e7184b38529669c5b97f62e7R140